core/src/Kokkos_View.hpp

```diff
@@ -2396,6 +2396,120 @@ KOKKOS_INLINE_FUNCTION DeducedCommonPropsType<Views...> common_view_alloc_prop(
+         (
+             // same layout
+             std::is_same<typename DstViewType::traits::array_layout,
+                          typename SrcViewType::traits::array_layout>::value ||
+             // known layout
+             ((std::is_same<typename DstViewType::traits::array_layout,
+                            Kokkos::LayoutLeft>::value ||
+               std::is_same<typename DstViewType::traits::array_layout,
+                            Kokkos::LayoutRight>::value ||
+               std::is_same<typename DstViewType::traits::array_layout,
+                            Kokkos::LayoutStride>::value) &&
+              (std::is_same<typename SrcViewType::traits::array_layout,
+                            Kokkos::LayoutLeft>::value ||
+               std::is_same<typename SrcViewType::traits::array_layout,
+                            Kokkos::LayoutRight>::value ||
+               std::is_same<typename SrcViewType::traits::array_layout,
+                            Kokkos::LayoutStride>::value)))),
+        void>::type> {
+ private:
+  using DstTraits = typename DstViewType::traits;
+  using SrcTraits = typename SrcViewType::traits;
+  using Mapping   = Kokkos::Impl::ViewMapping<DstTraits, SrcTraits,
+                                            typename DstTraits::specialize>;
+
+ public:
+  enum : bool { value = Mapping::is_assignable };
+};
+
+// -----------------------------------------------------------
+// public is_assignable
+// expose underlying view mapping properties case where layout isn't stride
+template <class DstViewType, class SrcViewType>
+inline static typename std::enable_if<
+    (!(std::is_same<typename SrcViewType::traits::array_layout,
+                    Kokkos::LayoutStride>::value) &&
+     // Added to have a new specialization for SrcType of
+     // LayoutStride
+     // default mappings
+     std::is_same<typename DstViewType::traits::specialize, void>::value &&
+     std::is_same<typename SrcViewType::traits::specialize, void>::value &&
+     (
+         // same layout
+         std::is_same<typename DstViewType::traits::array_layout,
+                      typename SrcViewType::traits::array_layout>::value ||
+         // known layout
+         ((std::is_same<typename DstViewType::traits::array_layout,
+                        Kokkos::LayoutLeft>::value ||
+           std::is_same<typename DstViewType::traits::array_layout,
+                        Kokkos::LayoutRight>::value ||
+           std::is_same<typename DstViewType::traits::array_layout,
+                        Kokkos::LayoutStride>::value) &&
+          (std::is_same<typename SrcViewType::traits::array_layout,
+                        Kokkos::LayoutLeft>::value ||
+           std::is_same<typename SrcViewType::traits::array_layout,
+                        Kokkos::LayoutRight>::value ||
+           std::is_same<typename SrcViewType::traits::array_layout,
+                        Kokkos::LayoutStride>::value)))),
+    bool>::type
```

Comment:
Don't copy and paste this from above. Move it off into a helper class template or something, but don't repeat something this complicated.